lib/matplotlib/axes/_subplots.py

```diff
@@ -91,7 +118,12 @@ def get_geometry(self):
     # COVERAGE NOTE: Never used internally or from examples
     def change_geometry(self, numrows, numcols, num):
         """change subplot geometry, e.g., from 1,1,1 to 2,2,3"""
-        self._subplotspec = GridSpec(numrows, numcols)[num - 1]
+        if self.figure.get_constrained_layout():
+            gridspecfig = fig
+        else:
+            gridspecfig = None
```

Comment:
I don't see gridspecfig being used in this function.  And fig is not defined there, either.  And the comment suggests that maybe this function should be deprecated, unless someone knows what the use case is.

Reply:
I'll remove the gridspecfig stuff.  No idea about whether the fcn should be deprecated or not...